search/search_service/proxy/es_search_proxy.py

```diff
@@ -0,0 +1,316 @@
+        else:
+            http_auth = (user, password) if user else None
+            # doesn't this go against the whole point oh having a singleton pattern?
+            self.elasticsearch = Elasticsearch(host, http_auth=http_auth)
+
+
+    def _build_term_query(self, resource: Resource, query_term: str) -> Q:
+        """
+        Builds the query object for the inputed search term
+        """
+        if query_term == "" or not query_term:
+            # We don't want to create multi_match query for ""
+            # because it will result in no matches even with filters
+            return None
+
+        fields = []
+
+        # TODO make resources into an enum
+        if resource == Resource.TABLE:
+            fields = ["display_name^1000",
+                      "name.raw^75",
+                      "name^5",
+                      "schema^3",
+                      "description^3",
+                      "column_names^2",
+                      "column_descriptions",
+                      "tags",
+                      "badges",
+                      "programmatic_descriptions"]
+        elif resource == Resource.DASHBOARD:
+            fields = ["name.raw^75",
+                      "name^7",
+                      "group_name.raw^15",
+                      "group_name^7",
+                      "description^3",
+                      "query_names^3",
+                      "chart_names^2"]
+        elif resource == Resource.FEATURE:
+            fields = ["feature_name.raw^25",
+                      "feature_name^7",
+                      "feature_group.raw^15",
+                      "feature_group^7",
+                      "version^7",
+                      "description^3",
+                      "status",
+                      "entity",
+                      "tags",
+                      "badges"]
+        elif resource == Resource.USER:
+            fields = ["full_name.raw^30",
+                      "full_name^5",
+                      "first_name.raw^5",
+                      "last_name.raw^5",
+                      "first_name^3",
+                      "last_name^3",
+                      "email^3"]
+        else:
+            # TODO if you don't specify a resource match for all generic fields in the future
+            raise ValueError(f"no fields defined for resource {resource}")
+
+        return MultiMatch(query=query_term, fields=fields, type='most_fields')
+
+
+    def _build_filters(self, resource: Resource, filters: List[Filter]) -> List:
+        """
+        Builds the query object for all of the filters given in the search request
+        """
+        mapping = self.RESOUCE_TO_MAPPING.get(resource)
+
+        filter_queries: List = []
+
+        for filter in filters:
+            filter_name = mapping.get(filter.name) if mapping != None and mapping.get(filter.name) != None else filter.name
+
+            queries_per_term = [Q(WILDCARD_QUERY,  **{filter_name: term}) for term in filter.values]
```

Comment:
why not https://github.com/elastic/elasticsearch-dsl-py/blob/master/elasticsearch_dsl/query.py#L494

Reply:
I think I'll stick to my current approach with queries, because essentially I'd have to add a | or & between all queries and that seems a bit clunky as opposed to just giving the bool query the list of wildcard queries and letting it handle it

Reply:
If we need some default query then i would replace wildcard * with match_all query